typing/gprinttyp.ml

```diff
@@ -139,16 +139,29 @@ module Index: sig
     | Main of int
     | Synthetic of int
     | Named_subnode of { id:int; synth:bool; name:string }
+  type level_and_scope = { level:int; scope: int }
+  type 'a desc = {
+    id: 'a;
+    color: Decoration.color option;
+    desc: Types.type_desc;
+    lvl:level_and_scope;
+  }
```

Comment:
I think that these new types could be declared outside the `Index` module, as they are morally orthogonal to the index logic. (But the current choice is okay.)